packages/connectivity/android/src/main/java/io/flutter/plugins/connectivity/ConnectivityPlugin.java

```diff
@@ -5,54 +5,95 @@
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
 import io.flutter.plugin.common.MethodChannel.Result;
 import io.flutter.plugin.common.PluginRegistry.Registrar;
 
 /** ConnectivityPlugin */
-public class ConnectivityPlugin implements MethodCallHandler {
+public class ConnectivityPlugin implements MethodCallHandler, StreamHandler {
+  private final Activity activity;
   private final ConnectivityManager manager;
+  private BroadcastReceiver receiver;
 
   /** Plugin registration. */
   public static void registerWith(Registrar registrar) {
     final MethodChannel channel =
         new MethodChannel(registrar.messenger(), "plugins.flutter.io/connectivity");
-    channel.setMethodCallHandler(new ConnectivityPlugin(registrar.activity()));
+    final EventChannel eventChannel =
+        new EventChannel(registrar.messenger(), "plugins.flutter.io/connectivity_status");
+    ConnectivityPlugin instance = new ConnectivityPlugin(registrar.activity());
+    channel.setMethodCallHandler(instance);
+    eventChannel.setStreamHandler(instance);
   }
 
   private ConnectivityPlugin(Activity activity) {
-    manager = (ConnectivityManager) activity.getSystemService(Context.CONNECTIVITY_SERVICE);
+    this.activity = activity;
+    this.manager = (ConnectivityManager) activity.getSystemService(Context.CONNECTIVITY_SERVICE);
+  }
+
+  @Override
+  public void onListen(Object arguments, EventSink events) {
+    receiver = createReceiver(events);
+    activity.registerReceiver(receiver, new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION));
+  }
+
+  @Override
+  public void onCancel(Object arguments) {
+    activity.unregisterReceiver(receiver);
+    receiver = null;
+  }
+
+  private static String getNetworkType(int type) {
+    switch (type) {
+      case ConnectivityManager.TYPE_ETHERNET:
+      case ConnectivityManager.TYPE_WIFI:
+      case ConnectivityManager.TYPE_WIMAX:
+        return "wifi";
+      case ConnectivityManager.TYPE_MOBILE:
+      case ConnectivityManager.TYPE_MOBILE_DUN:
+        return "mobile";
+      default:
+        return "none";
+    }
   }
 
   @Override
   public void onMethodCall(MethodCall call, Result result) {
     if (call.method.equals("check")) {
       NetworkInfo info = manager.getActiveNetworkInfo();
       if (info != null && info.isConnected()) {
-        switch (info.getType()) {
-          case ConnectivityManager.TYPE_ETHERNET:
-          case ConnectivityManager.TYPE_WIFI:
-          case ConnectivityManager.TYPE_WIMAX:
-            result.success("wifi");
-            break;
-          case ConnectivityManager.TYPE_MOBILE:
-          case ConnectivityManager.TYPE_MOBILE_DUN:
-            result.success("mobile");
-            break;
-          default:
-            result.success("none");
-            break;
-        }
-      } else {
-        result.success("none");
+        result.success(getNetworkType(info.getType()));
       }
+      result.success("none");
```

Comment:
This should be in an `else` - otherwise you'll always follow a connected response with a none response.

Reply:
Done had caught this while testing but did not push the change to PR.